daisy_workflows/image_import/windows/translate.ps1

```diff
@@ -93,6 +93,11 @@ function Setup-NTP {
   Set-ItemProperty -Path $tzi_path -Name RealTimeIsUniversal -Value 1
 
   # Set up time sync...
+  # Ensure the time service isn't in a pending state; wait until it is not in a pending state.
+  while((Get-Service -Name "W32time").Status -match 'pending') {
```

Comment:
It looks like there are several possible statuses.  Without knowing _anything_ about Windows services, I'd presume you'd want to assert the service is running (rather than stopped, for example). 

Here's the path I took to find the docs:

* https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-service
* https://docs.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontroller
* https://docs.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontrollerstatus

Lastly, I see this utility method; is it available from PowerShell?

https://docs.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontroller.waitforstatus

Reply:
WaitForStatus is available, but it is only able to wait for a single state. This isn't bad however it means you need to be more specific as there are [4 pended states](https://docs.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontrollerstatus?view=dotnet-plat-ext-3.1#fields), so the state you're waiting for would need to match the corresponding pending state.

WaitForStatus is likely more useful when your application/script does a state changes in such a way were you don't inherit the WaitForStatus behavior and that you're only looking for one specific state.

The goal of this is to ensure the Start|Stop-Service cmdlets are successful and for that just need to make sure the service isn't in one of the pending states. So I'd prefer to keep this as it's simplier and accomplishes the same goal.

Reply:
makes sense!